src/Shared/Microsoft.AspNetCore.Razor.Test.Common/IntegrationTestAttributes.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Test.Common;
+
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+[XunitTestCaseDiscoverer("Microsoft.AspNetCore.Razor.Test.Common." + nameof(IntegrationFactDiscoverer), "Microsoft.AspNetCore.Razor.Test.Common")]
+internal class IntegrationTestFactAttribute : FactAttribute { }
```

Comment:
A bit surprised that `internal` worked here as xUnit typically requires that everything be `public`.